src/xdp/xsk.c

```diff
@@ -45,6 +45,7 @@ typedef struct _UMEM_MAPPING {
     MDL *Mdl;
     UCHAR *SystemAddress;
     DMA_LOGICAL_ADDRESS DmaAddress;
+    BOOLEAN DoNotUnlockMdlPages;
```

Comment:
can we find a clearer name - why are the Mdl pages not supposed to be unlocked?

Reply:
can we just check if the caller is kernel?

Reply:
@nigriMSFT ?

Reply:
Yeah the naming can be clearer. There is a code comment where this field is set to help explain, but maybe it is more self-documenting with a better name

Reply:
Ah, I think I went with an explicit field in the UMEM_MAPPING, since the code path where this information is used does not have access to the XSK, only the UMEM, see XskDereferenceUmem

Reply:
I see, that makes sense, so I think the naming would be clearest if we negate the meaning of the current boolean field and call it `MdlPagesLocked` instead.